app/models/manageiq/providers/vmware/inventory/collector/network_manager.rb

```diff
@@ -0,0 +1,152 @@
+class ManageIQ::Providers::Vmware::Inventory::Collector::NetworkManager < ManageIQ::Providers::Vmware::Inventory::Collector
+  attr_reader :network_name_mapping
+  VappNetwork = Struct.new(:id, :name, :type, :is_shared, :gateway, :dns1, :dns2, :netmask, :enabled, :dhcp_enabled)
+
+  def orgs
+    return @orgs if @orgs.any?
+    @orgs = connection.organizations
+  end
+
+  def vdc_networks
+    return @vdc_networks if @vdc_networks.any?
+    @vdc_networks = orgs.each_with_object([]) do |org, res|
+      res.concat(org.networks.all)
+    end
+  end
+
+  def vapp_networks
+    return @vapp_networks if @vapp_networks.any?
+    @vapp_networks = manager.orchestration_stacks.each_with_object([]) do |stack, res|
+      fetch_network_configurations_for_vapp(stack.ems_ref).map do |net_conf|
+        # 'none' is special network placeholder that we must ignore
+        next if net_conf[:networkName] == 'none'
+
+        $vcloud_log.debug("#{log_header} processing net_conf for vapp #{stack.ems_ref}: #{net_conf}")
+        network_id = network_id_from_links(net_conf)
+        $vcloud_log.debug("#{log_header} calculated vApp network id: #{network_id}")
+        if (vdc_net = corresponding_vdc_network(net_conf, vdc_networks_idx))
+          $vcloud_log.debug("#{log_header} skipping VDC network duplicate")
+          memorize_network_name_mapping(stack.ems_ref, vdc_net.name, vdc_net.id)
+        else
+          memorize_network_name_mapping(stack.ems_ref, net_conf[:networkName], network_id)
+          res << build_vapp_network(stack, network_id, net_conf)
+
+          # routers connecting vApp networks to VDC networks
+          if (parent_net = parent_vdc_network(net_conf, vdc_networks_idx))
+            $vcloud_log.debug("#{log_header} connecting router to parent: #{parent_net}")
+            @network_routers << {
+              :net_conf   => net_conf,
+              :network_id => network_id,
+              :parent_net => parent_net
+            }
+          end
+        end
+      end
+    end
+  end
+
+  def nics
+    return @nics if @nics.any?
+    @nics = manager.vms.each_with_object([]) do |vm, res|
```

Comment:
hm, so in other providers, we usually just do the API query again. So the code is reusable for the targeted refresh. Maybe we can modify that, once you will be doing targeted refresh. (for targeted refresh of the Vm, we will want both Vm and interface in 1 refresh)